### PR TITLE
fix(core): ExperientialMemory.query neighbor weights use real score_sum for tiny positive sum

### LINKED ISSUE #253: ExperientialMemory.query gives tiny positive neighbor scores a near-zero weight instead of 1.0
## What's wrong

\`ExperientialMemory.query\` normalizes neighbor scores with:

\`\`\`python
neighbor_weights = positive_scores / jnp.maximum(score_sum, 1.0e-12)
\`\`\`

The \`1e-12\` floor is meant to guard the zero-score case (avoid dividing by zero when no neighbor is accepted), but it applies unconditionally rather than only when \`score_sum\` is exactly zero. Empirically, for a genuinely small-but-nonzero \`score_sum\` (e.g. a single accepted neighbor whose true similarity is exp(-36) ≈ 2.32e-16), the denominator floor produced a weight of approximately \`2.32e-4\` rather than \`1.0\` — the weights no longer summed to one for that neighbor, corrupting the weighted average over recovered actions.

Reproduced and verified via a real mutation-resistance transcript:

\`\`\`text
red (source fix reverted, regression test kept):
Mismatch at index [0]: 0.00023195227549877018 (ACTUAL), 1.0 (DESIRED)

green (fix restored):
47 passed
\`\`\`

## Reachability

\`ExperientialMemory\` is genuinely package-root exported (\`alberta_framework/__init__.py\` and \`alberta_framework/core/__init__.py\`, both import and \`__all__\`). \`ExperientialMemory.query\` is called by \`ExperientialMemoryPolicy.propose\`, which the public \`PrototypeAgent\`'s opt-in experiential-memory transaction calls via \`policy.propose\` — reachable through exported APIs with ordinary, configuration-valid inputs, not a hardcoded-safe-only path.

## Fix

Use the real \`score_sum\` as the denominator whenever it's positive, and fall back to \`1.0\` only when it's exactly zero (\`jnp.where(score_sum > 0.0, score_sum, 1.0)\`), rather than an unconditional epsilon floor via \`jnp.maximum\`.

PR incoming with a regression test using a real query at a configuration-valid sub-floor similarity, plus the mutation-resistance transcript.

### PR DESCRIPTION
**Pull request by sharkwon**
## Summary

Fixes #253 — ExperientialMemory.query gives tiny positive neighbor scores a near-zero weight instead of 1.0.

## Changes

- `alberta_framework/core/experiential_memory.py`: Replace unconditional `1e-12` floor with `jnp.where(score_sum > 0.0, score_sum, 1.0)` in neighbor weight normalization
- `tests/test_experiential_memory_253.py`: New regression test (single accepted neighbor at sub-floor similarity)

## Validation behavior

| Input | Before | After |
|-------|--------|-------|
| `score_sum == 0.0` | 1e-12 floor → weight ~1.0 | 1.0 fallback → weight 1.0 ✓ |
| `score_sum == 2.3e-16` (exp(-36)) | **1e-12 floor → weight ~2e-4** | **real score_sum → weight ~1.0** ✓ |
| `score_sum == 1.0` | 1.0 | 1.0 ✓ |

## Checks

- `pytest tests/test_experiential_memory.py tests/test_experiential_memory_policy.py` — 74/74 pass
- `pytest tests/test_experiential_memory_253.py` — 1/1 pass
- `ruff check` — clean

### CODE REVIEW

**Comment by lalalune** [comment]
Closing as a duplicate of merged PR #254. Current `main` already uses the real positive `score_sum` with an exact-zero fallback in `ExperientialMemory.query`, and it retains the focused regression coverage. Rebasing this branch produces the same source change plus a redundant standalone test module, so there is no additional production delta to merge.
